browser/brave_browser_process_impl.cc

```diff
@@ -153,6 +155,14 @@ void BraveBrowserProcessImpl::Init() {
 #endif
 
   InitSystemRequestHandlerCallback();
+
+#if !BUILDFLAG(IS_ANDROID)
+  if (!ObsoleteSystem::IsObsoleteNowOrSoon()) {
+    // Clear to show unsupported warning infobar again even if user
+    // suppressed it from previous os.
+    local_state()->ClearPref(prefs::kSuppressUnsupportedOSWarning);
+  }
+#endif
```

Comment:
Is this code for the case where obsolete OS becomes supported later on?

Reply:
When user dismissed this warning, that user could upgrade their os later.
And that os could be obsolete later. When it happens, this warning infobar should be visible again.
So, clear it here.

Reply:
> that user could upgrade their os later.

Oh, great 👍 Thanks.